src/live_restore/live_restore_fs.c

```diff
@@ -1194,46 +1194,59 @@ __live_restore_fs_open_in_source(WTI_LIVE_RESTORE_FS *lr_fs, WT_SESSION_IMPL *se
     return (ret);
 }
 
+/*
+ * __live_restore_compute_nbits --
+ *     Compute the number of bits needed for the bitmap, based off the destination file size.
+ */
+static int
+__live_restore_compute_nbits(
+  WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh, uint64_t *nbitsp)
+{
+    wt_off_t size;
+    WT_RET(lr_fh->destination->fh_size(lr_fh->destination, (WT_SESSION *)session, &size));
+    WT_ASSERT_ALWAYS(session, size % lr_fh->allocsize == 0,
+      "The file size isn't a multiple of the file allocation size!");
+    *nbitsp = (uint64_t)size / (uint64_t)lr_fh->allocsize;
+    return (0);
+}
+
 /*
  * __live_restore_decode_bitmap --
  *     Decode a bitmap from a hex string.
  */
 static int
-__live_restore_decode_bitmap(WT_SESSION_IMPL *session, const char *bitmap_str, uint64_t nbits,
-  WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh)
+__live_restore_decode_bitmap(WT_SESSION_IMPL *session, const char *bitmap_str,
+  uint64_t metadata_nbits, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh)
 {
     WT_DECL_RET;
     WT_ASSERT_ALWAYS(session, bitmap_str != NULL, "Live restore bitmap string is NULL");
 
-    WT_RET(__bit_alloc(session, nbits, &lr_fh->bitmap));
-    lr_fh->nbits = nbits;
+    WT_RET(__bit_alloc(session, metadata_nbits, &lr_fh->bitmap));
+    lr_fh->nbits = metadata_nbits;
 
     WT_ITEM buf;
     WT_CLEAR(buf);
     WT_ERR(__wt_hex_to_raw(session, bitmap_str, &buf));
     memcpy(lr_fh->bitmap, buf.mem, buf.size);
-    /* FIXME-WT-13813: Add check here for shorter dest than bitmap, set end to 1. */
+
+    uint64_t file_size_nbits;
+    WT_ERR(__live_restore_compute_nbits(session, lr_fh, &file_size_nbits));
```

Comment:
In theory we could shorten the bitmap here so we don't keep doing this but that feels overkill for now.